src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,34 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier{
+    Random random = new Random();
+    ColorSupplier colorSupplier = new ColorSupplier();
+    public Figure getRandomFigure(Figure[] figures){
+        int index = random.nextInt(figures.length);
+        if(index == 0){
+            Figure circle = new Circle(random.nextDouble(20), colorSupplier.getRandomColor());
```

Comment:
Creating a figure, don't pass expressions in the constructor. Create separate variables and pass them on for better code readability. Also, all magic numbers in your code should be constants.

Reply:
use switch statement instead
use Ctrl Alt L throughout your files